### PR TITLE
Test for writing after an lseek past end of file

### PR DESCRIPTION
**Pull request by RobbKistler**
This is a test to demonstrate Issue #375

### CODE REVIEW

File: test/integration-test-main.sh [hunk 1]
```diff
@@ -367,6 +367,13 @@ function test_rm_rf_dir {
    fi
 }
 
+function test_write_after_seek_ahead {
+   describe "Test writes succeed after a seek ahead"
+   ${TEST_SCRIPT_DIR}/test_helper write-after-seek-ahead testfile
```
Comment (andrewgaul) [inline test/integration-test-main.sh]: Can you replace `test_helper` with:

```
dd if=/dev/zero of=test seek=1 count=1 bs=1024
```

**Comment by RobbKistler** [comment]
Good idea.  dd fails when it does ftruncate().  It's likely exercising the same bug in s3fs-fuse.  Would it be OK to have both tests?  Or is test_helper.c too much build complication?

**Comment by andrewgaul** [comment]
I prefer to have all the tests in a single language.  Since `dd` can express this test and other options like `notrunc` more succinctly, could you rewrite this in bash?

**Comment by ggtakec** [comment]
@RobbKistler and @andrewgaul 
This PR has passed on Travis by merged #379.
When you have finished modification of test codes, please let me know.
After that, I will merge it soon.

Thanks in advance for your help.

**Comment by RobbKistler** [comment]
@andrewgaul I replaced the helper with dd.  If this looks good, I'll squash the commits.

**Comment by andrewgaul** [comment]
:+1:

**Comment by RobbKistler** [comment]
@ggtakec This PR is ready for merge. Thanks again for fixing the bug.

**Comment by ggtakec** [comment]
@RobbKistler thanks.

**Pull request merged**